service/controller/resource/service/create.go

```diff
@@ -21,19 +22,25 @@ func (r *Resource) ApplyCreateChange(ctx context.Context, obj, createChange inte
 	}
 
 	if serviceToCreate != nil {
-		r.logger.LogCtx(ctx, "level", "debug", "message", "creating Kubernetes service")
+		r.logger.LogCtx(ctx, "level", "debug", "message", "creating service")
 
 		namespace := key.ClusterNamespace(cr)
 		_, err = r.k8sClient.CoreV1().Services(namespace).Create(serviceToCreate)
 		if apierrors.IsAlreadyExists(err) {
 			// fall through
+		} else if apierrors.IsNotFound(err) {
+			r.logger.LogCtx(ctx, "level", "debug", "message", "did not create service")
+			r.logger.LogCtx(ctx, "level", "debug", "message", "namespace not found yet")
+			r.logger.LogCtx(ctx, "level", "debug", "message", "canceling resource")
+			resourcecanceledcontext.SetCanceled(ctx)
```

Comment:
Since `ApplyCreateChange` is the last method being executed anyway it does not make a difference to cancel the resource actively. Though I want to keep the pattern consistent with the logging and do it anyway.